python/tests/api/store/test_sqlite_store.py

```diff
@@ -0,0 +1,129 @@
+import os
+import sqlite3
+from datetime import datetime, timedelta, timezone
+from typing import List
+
+import pytest
+
+from whylogs.api.store.query import DatasetIdQuery, DateQuery
+from whylogs.api.store.sqlite_store import SQLiteStore
+from whylogs.core import DatasetProfile, DatasetProfileView
+
+
+class TestSQLiteStore:
+    @classmethod
+    def setup_class(cls):
+        os.environ["SQLITE_STORE_LOCATION"] = ":memory:"
+        os.environ["MERGE_PROFILE_PERIOD_HOURS"] = "1"
+
+    @classmethod
+    def teardown_class(cls):
+        del os.environ["MERGE_PROFILE_PERIOD_HOURS"]
+        del os.environ["SQLITE_STORE_LOCATION"]
+
+    @pytest.fixture
+    def store(self, profile_view):
+        store = SQLiteStore()
+        store.cur.execute("CREATE TABLE IF NOT EXISTS profile_store(id string, date datetime, profile blob);")
+        query = "INSERT INTO profile_store (id, date, profile) VALUES (?, ?, ?);"
```

Comment:
This seems weird to me. I don't think this thing should expose the ability to directly execute sql, and we shouldn't need to execute sql to test anything that its doing.

Reply:
in here i am avoiding to execute the store methods to insert stuff onto the store using the same connection, but at the same time populating its database before actually making it available to the tests.
making this to be able to compare expected vs. actual on `store.list`, and i dont want to `store.write` to later assert `store.list` returns me what it has written -- otherwise those wouldn't be unit tests

lmk how i could do this differently or better though

Reply:
Why do you want to avoid using the store to populate the store? That's a thing that it should be able to do. The alternative to that is copy/pasting the table creation query into the unit test (and re-copy/pasting it whenever we change the query in the store). It has to match exactly in order to work at all and its invalid for an end user to touch the sql directly. Here are the properties that don't work here:

- User being able to directly execute SQL.
- Tests having to exactly duplicate, by copy/pasting, table creation logic and table schema
- Tests having to exactly duplicate insert statements

If the store just ensures that tables exist after its made and exposes the appropriate methods for inserting things (which I think are both true) then we aren't losing anything. I know its Python so we probably just have to slap a few underscores infront of the conn/cur vars, though I am tempted to use scope to make it really private like 


```python
def make_hi():
  private_var = 'var'
  class Hi:
    def do_stuff():
      print(private_var)

  return Hi()

hi = make_hi()
hi.do_stuff()
```

That might be an anti pattern.

Reply:
> Why do you want to avoid using the store to populate the store? That's a thing that it should be able to do.

In order to make the tests actually unitary, I didn't want to have a "recursion" with the fixtures. 
So if I for some reason break or modify the way `store.write` works, `store.list` remains intact in the way it is tested and will continue to pass on the checks, and it may be easier to make the error be pointed out to where it's failing - if it ever comes to this point. But maybe that's wrong, that's just how I was told to think when creating unit tests in general. I might have taken it too literally on this. 

> It has to match exactly in order to work at all and its invalid for an end user to touch the sql directly.

With all that being said, all of this makes a ton of sense. Definitely needs improvement.